### ISSUE TITLE
Improper use of validation data and evaluation data in the Examples

### ISSUE THREAD

**Issue by icoxfog417** [fchollet/keras#5807]
I think evaluation data have to be independent from training and validation data.

* [How do you use test data set after Cross-validation](http://stats.stackexchange.com/questions/152907/how-do-you-use-test-data-set-after-cross-validation)
* [What is the difference between test set and validation set?](http://stats.stackexchange.com/questions/19048/what-is-the-difference-between-test-set-and-validation-set)

But in the some examples, **validation data is used as evaluation data**.

[keras/examples/imdb_lstm.py](https://github.com/fchollet/keras/blob/master/examples/imdb_lstm.py)

```
model.fit(x_train, y_train, batch_size=batch_size, epochs=15,
          validation_data=(x_test, y_test))
score, acc = model.evaluate(x_test, y_test,
                            batch_size=batch_size)
```

I think **we have to use [`validation_split`](https://github.com/fchollet/keras/blob/master/keras/models.py#L776) to make evaluation data be independent**.

```
model.fit(x_train, y_train, batch_size=batch_size, epochs=15, validation_split=0.2)
score, acc = model.evaluate(x_test, y_test,
                            batch_size=batch_size)
```

If this fix is reasonable, I'll make pull request.

**Comment by talpay** [comment]
We had a longer discussion on this in #1753, including some PRs but it (very unfortunately) went nowhere even as the fix is so simple. We were not able to convince that this is didactically bad practice and misleading to novices because even though any hyperparameter tuning will lead to overfitting on the test data, there is no overfitting if you don't touch any of the hyperparameters (which again, from my experience, is something almost every newcomer will do out of instinct).

**Comment by icoxfog417** [comment]
Thanks for your reply, and sorry for my poor search of issues.
I understand the discussion in #1753, and the `validation` in source code is not used for hyper parameter tuning. So the `validation` does not mean "hyper parameter tuning" but "checking the accuracy in the training". 

It is definitely confusing (it is proved by other issues and this), but I understood the meaning of validation in keras.

**Comment by talpay** [comment]
No, you're right that (cross-)validation is of course for hyperparameter tuning. That's universally true regardless of the framework. But again, it's difficult for me to argue against your proposal since I fully agree.
